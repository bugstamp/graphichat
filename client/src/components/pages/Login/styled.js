import { fadeInUp, fadeIn, headShake } from 'react-animations';
import styled, { keyframes } from 'styled-components';
import { backgrounds } from 'polished';

import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';

import blue from '@material-ui/core/colors/blue';

import { getStyledProps, getSpacing } from '../../../styles';
import bgImage from '../../../assets/images/login-bg__1920_65.jpg';

const fadeInUpAnimation = keyframes`${fadeInUp}`;
const fadeInAnimation = keyframes`${fadeIn}`;
const headShakeAnimation = keyframes`${headShake}`;

export const LoginWrapper = styled(Grid)`
  flex: 1 auto;
  display: flex;
  position: relative;
  ${backgrounds(`url(${bgImage})`, 'no-repeat')};
  background-size: cover;
  background-position: center;
`;

export const BrandTitleWrapper = styled.div`
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  margin-bottom: ${getSpacing(2)};
  animation: 1s ${fadeInUpAnimation};
  animation-delay: .5s;
  animation-fill-mode: forwards;
  opacity: 0;

  p {
    margin-left: ${getSpacing(1)};
    font-weight: bold;
    background: linear-gradient(to right, ${blue[100]} 0%, ${blue[500]} 100%);
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
  }
`;


export const LoginPresentationWrapper = styled.div`
  flex: 1 auto;
  display: flex;
  flex-flow: column;
  align-items: center;
  justify-content: center;
`;

export const SubTitle = styled.div`
  animation: 1s ${fadeInAnimation};
  animation-delay: 1s;
  animation-fill-mode: forwards;
  color: #fff;
  padding: 0 ${getSpacing(1)};
  text-align: center;
  opacity: 0;

  p {
    font-weight: bold;

  ${(props) => {
    const breakpoints = getStyledProps('theme.breakpoints')(props);
    const smDown = breakpoints.down('sm');
    const fontSize = getStyledProps('theme.typography.h5.fontSize')(props);

    return `
      ${smDown} {
        font-size: ${fontSize};
      }
    `;
  }};
  }

  button {
    display: inline-block;
    position: relative;
    border: none;
    outline: none;
    color: inherit;
    background-color: transparent;
    cursor: pointer;
    z-index: 0;
    -webkit-transform: translate3d(0,0,0);
    animation-name: ${({ stopAnimation }) => (!stopAnimation && headShakeAnimation)};
    animation-duration: 1.5s;
    animation-delay: 2.5s;
    animation-iteration-count: infinite;

    &:hover {
      color: ${getStyledProps('theme.palette.primary.main')}
    }

    p {
      display: inherit;
    }
  }
`;

export const LoginFormWrapper = styled.div`
  && {
    width: 100%;
    height: 100%;
    max-height: 100%;
    max-width: 375px;
    min-width: 320px;
    position: relative;
    padding: ${getSpacing(5)} ${getSpacing(3)};
    overflow: hidden auto;

  ${(props) => {
    const breakpoints = getStyledProps('theme.breakpoints')(props);
    const spacing = getStyledProps('theme.spacing')(props);
    const xsDown = breakpoints.down('xs');

    return `
      ${xsDown} {
        max-width: 100%;
        padding: ${spacing(2)}px;
      }
    `;
  }}
  }
`;

export const Header = styled(Typography)`
  width: 100%;
  position: relative;
  text-align: center;
`;

export const SignUpButton = styled(Button)`
  && {
    margin-top: 1em;
  }
`;
