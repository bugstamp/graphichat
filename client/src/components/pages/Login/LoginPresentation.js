import React, { memo } from 'react';
import PropTypes from 'prop-types';

import Typography from '@material-ui/core/Typography';

import BrandTitle from './BrandTitle';
import { LoginPresentationWrapper, SubTitle } from './styled';

const LoginPresentation = ({ stopAnimation, toggleForm }) => (
  <LoginPresentationWrapper>
    <BrandTitle />
    <SubTitle stopAnimation={stopAnimation}>
      <Typography component="p" variant="h4" align="center" paragraph>
        A lightweight, simple and useful web chat app based on the modern GraphQL API
      </Typography>
      <button type="button" onClick={toggleForm}>
        <Typography component="p" variant="h4" align="center">Try it now!</Typography>
      </button>
    </SubTitle>
  </LoginPresentationWrapper>
);

LoginPresentation.propTypes = {
  stopAnimation: PropTypes.bool.isRequired,
  toggleForm: PropTypes.func.isRequired,
};

export default memo(LoginPresentation);
