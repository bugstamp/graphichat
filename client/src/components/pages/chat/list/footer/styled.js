import styled from 'styled-components';

import { getSpacing } from '../../../../../styles';

export const ChatListFooterStyled = styled.div`
  display: flex;
  justify-content: center;
  padding: ${getSpacing(2)} ${getSpacing(1)};

  && {
    button {
      width: 50%;
    }
  }
`;

export default null;
