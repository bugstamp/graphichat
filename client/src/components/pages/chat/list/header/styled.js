import styled from 'styled-components';
import { position } from 'polished';

import IconButton from '@material-ui/core/IconButton';

import { getSpacing } from '../../../../../styles';

export const ChatListHeaderStyled = styled.div`
  display: flex;
  flex-flow: column;
  padding: ${getSpacing(2)} ${getSpacing(1)};
`;

export const HeaderTitle = styled.div`
  position: relative;
  margin-bottom: ${getSpacing(1)};
`;

export const AddButton = styled(IconButton)`
  && {
    ${position('absolute', 0, 0, null, null)};
    padding: ${getSpacing(0.5)};
  }
`;
