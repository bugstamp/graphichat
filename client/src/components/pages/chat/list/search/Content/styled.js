import styled from 'styled-components';

import DialogContent from '@material-ui/core/DialogContent';

import { SearchBoxStyled } from '../../../../../common/SearchBox/styled';
import { getStyledProps, getSpacing } from '../../../../../../styles';

export const SearchDialogContentStyled = styled(DialogContent)`
  && {
    flex: 1 auto;
    display: flex;
    flex-flow: column;
    padding: ${getSpacing(2)} ${getSpacing(3)};
    padding-bottom: 0;

    ${SearchBoxStyled} {
      margin-bottom: ${getSpacing(2)}
    }

  ${(props) => {
    const breakpoints = getStyledProps('theme.breakpoints')(props);
    const spacing = getStyledProps('theme.spacing')(props);
    const smDown = breakpoints.down('sm');

    return `
      ${SearchBoxStyled} {
        margin-bottom: ${getSpacing(1)}
      }
      ${smDown} {
        padding: ${spacing(1)}px ${spacing(2)}px;
      }
    `;
  }}
  }
`;

export default null;
