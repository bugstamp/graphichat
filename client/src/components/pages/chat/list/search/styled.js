import styled from 'styled-components';

import { getStyledProps } from '../../../../../styles';

export const SearchDialogBodyStyled = styled.div`
  && {
    display: flex;
    flex-flow: column;
    background-color: #fff;

  ${(props) => {
    const breakpoints = getStyledProps('theme.breakpoints')(props);
    const borderRadius = getStyledProps('theme.shape.borderRadius')(props);
    const shadows = getStyledProps('theme.shadows[1]')(props);
    const mdUp = breakpoints.up('md');

    return `
      ${mdUp} {
        min-height: 50vmin;
        max-height: 90vmin;
        border-radius: ${borderRadius}px;
        box-shadow: ${shadows};
      }
    `;
  }}
  }
`;

export default null;
