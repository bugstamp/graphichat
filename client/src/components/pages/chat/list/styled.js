import styled from 'styled-components';

import Paper from '@material-ui/core/Paper';

import { getStyledProps } from '../../../../styles';

export const ChatListStyled = styled(Paper)`
  && {
    position: relative;
    height: 100%;
    display: flex;
    flex-flow: column;
    background-color: ${getStyledProps('theme.palette.background.default')};
    border-left: 1px solid ${getStyledProps('theme.palette.grey.200')};
    border-right: 1px solid ${getStyledProps('theme.palette.grey.200')};
  }
`;
