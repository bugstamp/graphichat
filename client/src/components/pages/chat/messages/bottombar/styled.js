import styled from 'styled-components';

import Input from '@material-ui/core/Input';
import Fab from '@material-ui/core/Fab';
import IconButton from '@material-ui/core/IconButton';

import { ListItemAvatarStyled } from '../../../../common/List/ListItemAvatar';

import { getStyledProps, getSpacing } from '../../../../../styles';

export const ChatBottomBarStyled = styled.div`
  min-height: 72px;
  display: flex;
  flex-flow: row nowrap;
  position: relative;
  border-top: 1px solid ${getStyledProps('theme.palette.grey.200')};
  background-color: #fff;

  ${ListItemAvatarStyled} {
    padding: ${getSpacing(1)};
  }
`;

export const Form = styled.form`
  width: 100%;
  height: 100%;
  display: flex;
  position: relative;
  padding: ${getSpacing(1)} 0;

  ${(props) => {
    const breakpoints = getStyledProps('theme.breakpoints')(props);
    const smDown = breakpoints.down('sm');

    return `
      ${smDown} {
        min-height: auto;
        flex-flow: row nowrap;
        align-items: center;
        padding: ${getSpacing(0.5)(props)};
      }
    `;
  }}
`;

export const FormInput = styled(Input)`
  && {
    flex: 1 auto;
    display: flex;
    align-items: flex-start;

  ${(props) => {
    const breakpoints = getStyledProps('theme.breakpoints')(props);
    const spacing = getStyledProps('theme.spacing')(props);
    const smDown = breakpoints.down('sm');

    return `
      ${smDown} {
        padding: ${spacing(1)}px;
        border: 1px solid ${getStyledProps('theme.palette.grey.400')(props)};
        border-radius: 20px;

        &:before,
        &:after {
          display: none;
        }
      }
    `;
  }}
  }
`;

export const SubmitButton = styled(IconButton)`
  && {
    padding: ${getSpacing(1)};
  }
`;

export const FabSubmitButton = styled(Fab)`
  && {
    position: absolute;
    top: -24px;
    right: 0;
    z-index: 30;
  }
`;
