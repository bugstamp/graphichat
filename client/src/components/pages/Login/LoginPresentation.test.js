import React from 'react';
import { shallow } from 'enzyme';
import Typography from '@material-ui/core/Typography';

import LoginPresentation from './LoginPresentation';
import { SubTitle } from './styled';

import { mountMockedProvider } from '../../../__mocks__/mockedProvider';

describe('LoginPresentation', () => {
  const stopAnimation = false;
  const toggleFormMock = jest.fn();

  const LoginPresentationMock = (
    <LoginPresentation
      stopAnimation={stopAnimation}
      toggleForm={toggleFormMock}
    />
  );
  const mountWrapper = () => mountMockedProvider(LoginPresentationMock);

  test('should match to snapshot', () => {
    const wrapper = shallow(LoginPresentationMock);

    expect(wrapper).toMatchSnapshot();
  });
  test('mount should be without errors', () => {
    const wrapper = mountWrapper();

    expect(wrapper.find(LoginPresentation)).toHaveLength(1);
  });
  test('passed props should be match', () => {
    const wrapper = mountWrapper();

    expect(wrapper.find(LoginPresentation).prop('stopAnimation')).toBe(stopAnimation);
    expect(wrapper.find(LoginPresentation).prop('toggleForm')).toBe(toggleFormMock);
  });
  test('should have BrandTitle', () => {
    const wrapper = mountWrapper();

    expect(wrapper.find('BrandTitle')).toHaveLength(1);
  });

  describe('SubTitle', () => {
    const wrapper = mountWrapper();
    const SubTitleWrapper = wrapper.find(SubTitle);

    it('should have stopAnimation prop', () => {
      expect(SubTitleWrapper.prop('stopAnimation')).toBe(stopAnimation);
    });
    it('should have button with onClick prop', () => {
      expect(SubTitleWrapper.find('button')).toHaveLength(1);
      expect(SubTitleWrapper.find('button').prop('onClick')).toBe(toggleFormMock);
    });
    it('should have two headers with correct props', () => {
      const headers = SubTitleWrapper.find(Typography);

      expect(headers).toHaveLength(2);
      expect(SubTitleWrapper.find(Typography).first().props()).toMatchObject({
        component: 'p',
        variant: 'h4',
        align: 'center',
        paragraph: true,
        children: 'A lightweight, simple and useful web chat app based on the modern GraphQL API',
      });
      expect(SubTitleWrapper.find(Typography).last().props()).toMatchObject({
        component: 'p',
        variant: 'h4',
        align: 'center',
        children: 'Try it now!',
      });
    });
  });
});
