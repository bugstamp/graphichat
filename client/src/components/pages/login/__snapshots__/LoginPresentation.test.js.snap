// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPresentation should match to snapshot 1`] = `
<styled__LoginPresentationWrapper>
  <BrandTitle />
  <styled__SubTitle
    stopAnimation={false}
  >
    <WithStyles(ForwardRef(Typography))
      align="center"
      component="p"
      paragraph={true}
      variant="h4"
    >
      A lightweight, simple and useful web chat app based on the modern GraphQL API
    </WithStyles(ForwardRef(Typography))>
    <button
      onClick={[MockFunction]}
      type="button"
    >
      <WithStyles(ForwardRef(Typography))
        align="center"
        component="p"
        variant="h4"
      >
        Try it now!
      </WithStyles(ForwardRef(Typography))>
    </button>
  </styled__SubTitle>
</styled__LoginPresentationWrapper>
`;
